### PR TITLE
Raise exception on unknown format

### LINKED ISSUE #30: Implementations *must* raise UnsupportedFormatException on unknown formats
Consider this code

```
    try:
        carrier = {}
        span.tracer.inject(span, SOME_CUSTOM_FORMAT, carrier)
        do_something(carrier)
    except opentracing.UnsupportedFormatException:
        pass  # it's possible tracer does not support out custom format
    except:
        log.exception('Failed to inject tracing span into headers')
        metric.incCounter()
```

Without a mandate for the tracers (including the Noop tracer) to return an exception on unknown format there is no way in the code to know if the situation is legitimate (i.e. tracer does not support given custom format) or a genuine error (bad context propagation or what not).

### PR DESCRIPTION
**Pull request by yurishkuro**
Fixes #30

### CODE REVIEW

**Comment by bg451** [comment]
#31 seems to be a dupe, want to close that? 

Other than that, lgtm.

**Comment by yurishkuro** [comment]
Not a dupe, one for 1.0 another for 2.0

File: opentracing/tracer.py [hunk 1]
```diff
@@ -98,7 +99,9 @@ def inject(self, span_context, format, carrier):
             is defined by python `==` equality.
         :param carrier: the format-specific carrier object to inject into
         """
-        pass
+        if format == Format.TEXT_MAP or format == Format.BINARY:
```
Comment (bensigelman) [inline opentracing/tracer.py]: HTTP_HEADERS, too, per https://github.com/opentracing/opentracing-python/blob/f6b18a69177f11be6660f8346596cf0736f64674/opentracing/propagation.py#L93

File: opentracing/tracer.py [hunk 2]
```diff
@@ -123,7 +126,9 @@ def extract(self, format, carrier):
         :return: a SpanContext instance extracted from `carrier` or None if no
             such span context could be found.
         """
-        return self._noop_span_context
+        if format == Format.TEXT_MAP or format == Format.BINARY:
```
Comment (bensigelman) [inline opentracing/tracer.py]: (same)

**Comment by bensigelman** [comment]
LG aside from HTTP_HEADERS thing.

**Comment by bensigelman** [comment]
Nit: maybe also squash the commits down?

**Comment by yurishkuro** [comment]
good catch w/ http headers

Github can squash commits at the time of merge, if you press the right button.

**Comment by yurishkuro** [comment]
@bensigelman with http_headers and another test

**Comment by bensigelman** [comment]
🚢

**Pull request merged**